lib/xcodeproj/project/object/build_configuration.rb

```diff
@@ -79,16 +81,19 @@ def type
         #        the key of the build setting.
         #
         # @param [PBXNativeTarget] root_target
-        #        use this to resolve complete recursion between project and targets
+        #        use this to resolve complete recursion between project and targets.
+        #
+        # @param [String] previous_key
+        #        use this to resolve complete recursion between different build settings.
         #
         # @return [String] The value of the build setting
         #
-        def resolve_build_setting(key, root_target = nil)
+        def resolve_build_setting(key, root_target = nil, previous_key = nil)
```

Comment:
is it always sufficient to have a single previous key, or do we need a list of all of them?

Reply:
Do you mean for cases like this:

https://github.com/CocoaPods/Xcodeproj/blob/12cca46550a3634b5d062be1e3f62f867055d1c8/spec/project/object/build_configuration_spec.rb#L75-L76

?

Case in which both `mutually_recursive` and `mutually_recursive_nested` are resolved to `nil`:

https://github.com/CocoaPods/Xcodeproj/blob/12cca46550a3634b5d062be1e3f62f867055d1c8/spec/project/object/build_configuration_spec.rb#L122-L123

Reply:
Or if you have another specific case in mind, I could make a test for it and see if it works.